File: Medicare/src/main/java/com/service/CartService.java
package com.service;


import org.springframework.stereotype.Service;

import com.model.Cart;
import com.model.CartDetail;

@Service
public class CartService {

    private Cart cart;

    public CartService() {
        this.cart = new Cart();
    }

    public void addItem(CartDetail cartDetail) {
        cart.addItem(cartDetail);
    }

    public void removeItem(CartDetail cartDetail) {
        cart.removeItem(cartDetail);
    }

    public Cart getCart() {
        return cart;
    }

  
}

File: medicare_app/src/app/cart.service.ts
// cart.service.ts
import { Injectable } from '@angular/core';
import { Medicine } from './models/medicine.model';

@Injectable({
  providedIn: 'root',
})
export class CartService {
  private cart: Medicine[] = [];

  addToCart(medicine: Medicine): void {
    this.cart.push(medicine);
  }

  removeFromCart(medicine: Medicine): void {
    const index = this.cart.indexOf(medicine);
    if (index > -1) {
      this.cart.splice(index, 1);
    }
  }

  getCart(): Medicine[] {
    return this.cart;
  }

  // Other cart-related logic
}
